connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/BatchExecutableProcessor.java

```diff
@@ -78,14 +81,24 @@ public Map<UUID, RegisteredExecutable> activateBatch(
 
     for (var entry : request.entrySet()) {
       final UUID id = entry.getKey();
-      final InboundConnectorDetails data = entry.getValue();
+      final InboundConnectorDetails maybeValidData = entry.getValue();
+      final ValidInboundConnectorDetails data;
+
+      if (maybeValidData instanceof InvalidInboundConnectorDetails invalid) {
+        alreadyActivated.put(
+            id, new RegisteredExecutable.InvalidDefinition(invalid, invalid.error().getMessage()));
+        continue;
+      } else {
+        data = (ValidInboundConnectorDetails) maybeValidData;
+      }
 
       final RegisteredExecutable result =
           activateSingle(data, e -> cancellationCallback.accept(e, id));
 
       switch (result) {
         case Activated activated -> alreadyActivated.put(id, activated);
         case ConnectorNotRegistered notRegistered -> alreadyActivated.put(id, notRegistered);
+        case InvalidDefinition invalid -> alreadyActivated.put(id, invalid);
```

Comment:
I like how easily this new case fits into the existing switch, nice design work 😄